pyparsing/core.py

```diff
@@ -645,20 +605,21 @@ def _parseNoCache(self, instring, loc, doActions=True, callPreParse=True):
                             and isinstance(tokens, (ParseResults, list)),
                             modal=self.modalResults,
                         )
-        if debugging:
-            # ~ print("Matched", self, "->", retTokens.asList())
-            if self.debugActions[MATCH]:
-                self.debugActions[MATCH](instring, tokensStart, loc, self, retTokens)
+            except Exception as err:
+                self.debug and self.debugActions.FAIL(instring, start, self, err)
+                raise
+        self.debug and self.debugActions.MATCH(instring, start, loc, self, retTokens)
```

Comment:
renamed `tokenStart` to `start` so this line is a bit shorter and Black keeps it on this line.

Reply:
Would rather see as explicit:

    if self.debug:
        self.debugActions.FAIL(etc.)